packages/google_maps_flutter/google_maps_flutter_android/android/src/main/java/io/flutter/plugins/googlemaps/CozyMarkerBuilder.java

```diff
@@ -82,56 +42,141 @@ private float getTextXOffset(float markerWidth, Rect rect) {
         return (markerWidth / 2f) - (rect.width() / 2f) - rect.left;
     }
 
-    private Path addTailOnMarkerCenter(Bitmap marker, int tailSize, int shadowSize) {
+    private Path addPointerOnMarkerCenter(Bitmap marker, int pointerWidth, int pointerHeight, int shadowSize) {
         Path pointer = new Path();
         pointer.setFillType(Path.FillType.EVEN_ODD);
         float width = marker.getWidth();
-        float height = marker.getHeight() - tailSize - shadowSize;
-        pointer.moveTo(width / 2f - tailSize, height);
-        pointer.lineTo(width / 2f + tailSize, height);
-        pointer.lineTo(width / 2f, height + tailSize);
-        pointer.lineTo(width / 2f - tailSize, height);
+        float height = marker.getHeight() - pointerHeight - shadowSize;
+        pointer.moveTo(width / 2f - pointerWidth, height);
+        pointer.lineTo(width / 2f + pointerWidth, height);
+        pointer.lineTo(width / 2f, height + pointerHeight);
+        pointer.lineTo(width / 2f - pointerWidth, height);
         pointer.close();
         return pointer;
     }
 
-    private Bitmap getMarkerBitmap(String text, int markerColor, int textColor, boolean hasTail) {
+    private Bitmap getIconBitmap(String svgIcon, int width, int height) {
+        if(svgIcon == null)
+            return null;
+        
+        final Bitmap bitmap = markerCache.getBitmapFromMemCache(svgIcon);   
+        if (bitmap != null) {
+            return bitmap;
+        }
+
+        try{
+            //Needed this because svg doesn't scale if there is fixed width and height
+            String finalSvgIcon = svgIcon.replaceAll("width=\"24\" height=\"24\" ", "");
+            SVG  svg = SVG.getFromString(finalSvgIcon);
+            svg.setDocumentViewBox(0,0,24,24);
+
+            Bitmap iconBitmap = Bitmap.createBitmap(width, height, Bitmap.Config.ARGB_8888);
+            Canvas canvas = new Canvas(iconBitmap);
+            svg.renderToCanvas(canvas, new RectF(0, 0, width, height));
+
+            markerCache.addBitmapToMemoryCache(svgIcon, iconBitmap);
+            return iconBitmap;
+        }catch(Exception e){}
+
+        return null;
+    }
+
+    private Bitmap getMarkerBitmap(CozyMarkerData markerData) {
+        String text = markerData.label;
+        boolean hasPointer = markerData.hasPointer;
+
+        /* setting colors */
+        int defaultMarkerColor = Color.WHITE;
+        int defaultTextColor = Color.BLACK;
+        int defaultIconCircleColor = Color.rgb(248, 249, 245);
+
+        int selectedMarkerColor = Color.rgb(57, 87, 189);
+        int selectedTextColor = Color.WHITE;
+        int selectedIconCircleColor = Color.WHITE;
+
+        int visualizedMarkerColor = Color.rgb(248, 249, 245);
+        int visualizedTextColor = Color.rgb(110, 110, 100);
+        int visualizedIconCircleColor = Color.WHITE;
```

Comment:
What about defining these as consts? That way we either make sure they won't be modified and also differentiate them better from the variables that will hold the "chosen" colors below